src/main/kotlin/com/cjbooms/fabrikt/model/KotlinTypeInfo.kt

```diff
@@ -55,7 +58,12 @@ sealed class KotlinTypeInfo(val modelKClass: KClass<*>, val generatedModelClassN
         fun from(schema: Schema, oasKey: String = "", enclosingName: String = ""): KotlinTypeInfo =
             when (schema.toOasType(oasKey)) {
                 OasType.Date -> Date
-                OasType.DateTime -> DateTime
+                OasType.DateTime -> if (MutableSettings.typeOptions().contains(TypeCodeGenOptionType.INSTANT_DATETIME_TYPE)
```

Comment:
pull this out to a private function to aid readability, something like:
`getOverridableDateTimeType()`